### ISSUE TITLE
test_sanity_ocount_run fails

### ISSUE THREAD

**Issue by omron93** [sclorg/devtoolset-container#7]
test_sanity_ocount_run in devtoolset-4-perftools test file is failing on machines with older unsupported types of CPU. When I try it, with oprofile-0.9.9 it is working but with oprofile-1.0.0 it is failing with:

```
19:42:21 [1m[INFO] Testing 'ocount ls' usage ...[0m
19:42:21 Unable to obtain cpu_type
19:42:21 Verify that a pre-1.0 version of OProfile is not in use.
19:42:21 If the /dev/oprofile/cpu_type file exists, locate the pre-1.0 OProfile
19:42:21 installation, and use its 'opcontrol' command, passing the --deinit option.
19:42:21 cpu_type 'unset' is not valid
19:42:21 you should upgrade oprofile or force the use of timer mode
19:42:22 Default perf event: 
19:42:23 [1;31m[ERROR] Exit code is zero: FAIL (1)[0m
19:42:23 ocount output:
19:42:23 ----- ----- -----
19:42:23 Your kernel's Performance Events Subsystem does not support your processor type.
19:42:23 ----- ----- -----
19:42:23 [1;32m[PASS] at least,  should appear in report: PASS[0m
19:42:23 [1;31m[ERROR] Some tests failed[0m
19:42:23 make: *** [test] Error 1
```

@happz  Do you have any idea how to solve it? (this old machine cpu occurrence is for example from our Jenkins RHEL7 testing slave from OpenStack)

**Comment by frozenstein** [comment]
Hi, I am interested in some more details. Is the environment where 0.9.9 passes and 1.0.0 fails really the same? Could you provide more info about the CPU (`lscpu`, `/proc/cpuinfo`)? Does `perf stat -e cycles ls` work there (you need perf rpm package)? There might be also a problem with virtualization, since the host machine needs special setup to allow the guest to use the PMU. Also, if oprofile-0.9.9 works, could you please paste an output of `ophelp -r` command?

There were many changes between oprofile 0.9.9 and 1.0.0, e.g. removal of `opcontrol` and related legacy profiling mode, however I am not aware of any CPU support removal wrt `ocount`.

**Comment by omron93** [comment]
@frozenstein Sorry I was wrong. Both versions `ocount` returns 'Your kernel's Performance Events Subsystem does not support your processor type.'. I does not tested this, because I thought probles is in `ophelp -d` which is run firsts (https://github.com/sclorg/devtoolset-container/blob/master/4-perftools/test/run#L88) - in 1.0.0 this command fails (output in first comment), in 0.9.9 this command does not fail (has empty output :-)). So probably both versions does not support CPU :-(

Machine info:

```
#  lscpu
Architektura:          x86_64
Operační režim(y) CPU:32-bit, 64-bit
Pořadí bajtů:       Little Endian
Počet CPU:            1
Seznam zapnutých CPU: 0
Vláken na jádro:     1
Jader na patici:       1
Patic:                 1
Uzly NUMA:             1
ID výrobce:           GenuineIntel
Rodina CPU:            6
Model:                 60
Název modelu:         Intel Core Processor (Haswell, no TSX)
Stepping:              1
CPU MHz:               2599.996
BogoMIPS:              5199.99
Virtualizace:          VT-x
Výrobce hypervizoru:  KVM
Druh virtualizace:     plná
L1d keš:              32K
L1i keš:              32K
L2 keš:               4096K
CPU NUMA uzlu 0:       0
```

```
# perf stat -e cycles ls
jdk  slave.jar  workspace

 Performance counter stats for 'ls':

   <not supported>      cycles                   

       0,000838464 seconds time elapsed

```

> There might be also a problem with virtualization, since the host machine needs special setup to allow the guest to use the PMU

So is this our problem?

**Comment by frozenstein** [comment]
Well, your CPU is an Intel Haswell. These are well supported by oprofile from 0.9.9 above (also note that Red Hat oprofile package of version 0.9.9 contains patches backported from upstream 1.0 and 1.1; thus, it also supports models not supported by upstream 0.9.9). Since `perf stat -e cycles` gives `<not supported>`, I'd bet that the problem is in the virtualization configuration. You can use PMU events (~"oprofile events") within KVM guests, but you need to enable that. I am not an expert in KVM/QEMU. This might help, however, I am really not sure if it is enough: https://libvirt.org/formatdomain.html (search for "PMU").

The `perf stat -e cycles ls` command can be always used as a basic PMU test. If you manage to configure the KVM/QEMU correctly, this command will return a numeric value. It should be roughly the same as what `ocount ls` returns.

**Comment by frozenstein** [comment]
Also please note that the guest CPU should be inherited from the host. In guests, you can emulate models that differ from what the host runs on. If this mismatches, the events might give incorrect results, even if PMU is enabled correctly.

**Comment by omron93** [comment]
@frozenstein Thank you very much for deep explanation.

**Comment by hhorak** [comment]
Fixed via 4b45c3415.
